test/integration_test.go

```diff
@@ -289,37 +289,37 @@ func (m txManager) addAndWait(args ...txn.Arg) ([]byte, error) {
 
 		sentTxnID := sentTxn.GetID()
 
-	events:
-		for event := range events {
-			for _, result := range event.Transactions {
-				fetchedTxnID := result.GetTransaction().GetID()
+		accepted := isAccepted(events, sentTxnID)
+		if accepted {
+			return sentTxnID, nil
+		}
 
-				if bytes.Equal(sentTxnID, fetchedTxnID) {
-					accepted, status := event.Transactions[0].GetStatus()
+		err = m.m.Sync()
+		if err != nil {
+			return nil, xerrors.Errorf("failed to sync: %v", err)
+		}
 
-					if !accepted {
-						if i+1 == m.retry {
-							return nil, xerrors.Errorf("transaction not accepted: %s", status)
```

Comment:
no need for this timeout/max-retry anymore ?

Reply:
The mechanism is still there, but thanks to the refactoring no need for this ugly check anymore :)